core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.ts

```diff
@@ -44,12 +60,22 @@ export class DotExperimentsConfigurationVariantsComponent {
     maxVariantsAllowed = MAX_VARIANTS_ALLOWED;
     defaultVariantName = DEFAULT_VARIANT_NAME;
     experimentStepName = ExperimentSteps.VARIANTS;
-
     @Input() stepStatus: StepStatus;
     @Input() variants: Variant[];
-
     @Output() sidebarStatusChanged = new EventEmitter<SidebarStatus>();
     @Output() delete = new EventEmitter<Variant>();
+    @Output() edit = new EventEmitter<Pick<DotExperiment, 'name' | 'id'>>();
     @Output() save = new EventEmitter<Pick<DotExperiment, 'name'>>();
     @Output() goToEditPage = new EventEmitter<{ variant: Variant; mode: EditPageTabs }>();
+
+    constructor(private readonly cdr: ChangeDetectorRef) {}
+
+    editVariantName(newVariantName: string, variant: Variant, button: HTMLButtonElement) {
+        this.edit.emit({
+            ...variant,
+            name: newVariantName
+        });
+        button.disabled = true;
+        this.cdr.markForCheck();
```

Comment:
You need to the button attr in the template so you don't have to manually detect changes.